pkg/cfn/manager/api.go

```diff
@@ -243,38 +245,55 @@ func (c *StackCollection) DeleteStack(name string, force bool) (*Stack, error) {
 		}
 		return nil, err
 	}
-	if *s.StackStatus == cloudformation.StackStatusDeleteFailed && !force {
-		return nil, fmt.Errorf("stack %q previously couldn't be deleted", name)
-	}
 	i.StackId = s.StackId
+	return c.DeleteStackBySpec(i)
+}
+
+// DeleteStackByNameSync sends a request to delete the stack, and waits until status is DELETE_COMPLETE;
+// any errors will be written to errs channel, assume completion when nil is written, do not expect
+// more then one error value on the channel, it's closed immediately after it is written to
+func (c *StackCollection) DeleteStackByNameSync(name string, errs chan error) error {
+	i, err := c.DeleteStackByName(name)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("waiting for stack %q to get deleted", *i.StackName)
+
+	go c.waitUntilStackIsDeleted(i, errs)
+
+	return nil
+}
+
+// DeleteStackBySpec sends a request to delete the stack
+func (c *StackCollection) DeleteStackBySpec(s *Stack) (*Stack, error) {
```

Comment:
Why does this return a stack?

Reply:
So that we can pass it to `c.waitUntilStackIsDeleted`.